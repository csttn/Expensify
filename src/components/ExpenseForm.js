import React from "react";
import moment from "moment";

import { SingleDatePicker } from "react-dates";

export default class ExpenseForm extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      description: props.expense ? props.expense.description : "",
      note: props.expense ? props.expense.note : "",
      amount: props.expense ? (props.expense.amount / 100).toString() : "",
      createdAt: props.expense
        ? moment(props.expense.createdAt)
        : moment().locale("pt-br"),
      calendarFocused: false,
      error: ""
    };
  }

  onDescriptionChange = e => {
    const description = e.target.value;
    this.setState(() => ({ description }));
  };
  onNoteChange = e => {
    const note = e.target.value;
    this.setState(() => ({ note }));
  };
  onAmountChange = e => {
    const amount = e.target.value;

    if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
      this.setState(() => ({ amount }));
    }
  };
  onDateChange = createdAt => {
    if (createdAt) {
      this.setState(() => ({ createdAt }));
    }
  };
  onFocusChange = ({ focused }) => {
    this.setState(() => ({ calendarFocused: focused }));
  };

  onSubmit = e => {
    e.preventDefault();
    if (!this.state.description || !this.state.amount) {
      this.setState(() => ({
        error: "Por favor preencha os campos de valor e descrição"
      }));
    } else {
      this.setState(() => ({ error: "" }));
      this.props.onSubmit({
        description: this.state.description,
        amount: parseFloat(this.state.amount, 10) * 100,
        createdAt: this.state.createdAt.valueOf(),
        note: this.state.note
      });
    }
  };

  render() {
    return (
      <form onSubmit={this.onSubmit} className="form">
        {this.state.error && <p className="form__error">{this.state.error}</p>}
        <input
          type="text"
          placeholder="Descrição"
          autoFocus
          className="text-input"
          value={this.state.description}
          onChange={this.onDescriptionChange}
        />
        <input
          type="text"
          placeholder="Valor"
          value={this.state.amount}
          onChange={this.onAmountChange}
          className="text-input"
        />
        <SingleDatePicker
          date={this.state.createdAt}
          onDateChange={this.onDateChange}
          focused={this.state.calendarFocused}
          onFocusChange={this.onFocusChange}
          numberOfMonths={1}
          isOutsideRange={() => false}
        />
        <textarea
          className="textarea"
          placeholder="Adicione uma anotação sobre a sua despesa"
          value={this.state.note}
          onChange={this.onNoteChange}
        ></textarea>

        <div>
          <button className="box-layout__button">Salvar Despesa</button>
        </div>
      </form>
    );
  }
}
