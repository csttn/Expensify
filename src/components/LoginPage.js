import React from "react";
import { connect } from "react-redux";
import { startLogin } from "../actions/auth";

export const LoginPage = ({ startLogin }) => {
  return (
    <div className="box-layout">
      <div className="box-layout__box">
        <h1 className="box-layout__title">Expensify</h1>
        <p className="box-layout__subtitle">Hora de controlar suas despesas!</p>
        <button className="box-layout__button" onClick={startLogin}>
          Login com <span className="box-layout__button_span">Google</span>
        </button>
      </div>
    </div>
  );
};

const mapDispatchToProps = dispatch => ({
  startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
