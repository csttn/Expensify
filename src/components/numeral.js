import numeral from "numeral";

numeral.register("locale", "brasil", {
  delimiters: {
    thousands: ".",
    decimal: ","
  },
  abbreviations: {
    thousand: "mil",
    million: "milhões",
    billion: "b",
    trillion: "t"
  },
  ordinal: () => "º",
  currency: {
    symbol: "R$"
  }
});

numeral.locale("brasil");
export default numeral;
