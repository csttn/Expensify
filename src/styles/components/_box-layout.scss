.box-layout {
  align-items: center;
  background: url("/assets/bg3.jpg");
  background-size: cover;
  display: flex;
  justify-content: center;
  height: 100vh;
  width: 100vw;
}

.box-layout__box {
  background: fade-out($dark, 0.2);
  border-radius: 0.3rem;
  padding: $l-size $m-size;
  text-align: center;
  width: 25rem;
  
}

.box-layout__title {
  margin: 0 0 $m-size;
  line-height: 1;
  color: $rose;
}

.box-layout__subtitle {
    color: $gray;
}

.box-layout__button {
    font-size: 1.8rem; 
    background-color: $blue;
    color: $gray;
    border-radius: 0.2rem;
    border: none;
    padding: 1rem;
    font-weight: 500;
}

.box-layout__button_span {
    color: $orange;
    font-weight: 900;
    font-size: 2rem;
}