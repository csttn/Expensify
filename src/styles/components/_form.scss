.form {
  display: flex;
  flex-direction: column;
  > * {
    margin-bottom: $m-size;
  }
}

.form__error {
  margin: 0 0 $m-size 0;
  font-style: italic;
}

.button--remove {
  font-size: 1.8rem;
  color: $gray;
  border-radius: 0.2rem;
  border: none;
  padding: 1rem;
  font-weight: 500;
  background-color: $rose;
}
