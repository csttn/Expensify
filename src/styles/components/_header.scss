.header {
  background-color: $dark;
}

.header__content {
  align-items: center;
  margin: 0;
  display: flex;
  justify-content: space-between;
  padding: $s-size 0;
}

.header-title {
  color: $rose;
  margin: 0;
}

.header-link {
  text-decoration: none;
}

.header-button {
  background-color: $dark;
  color: $rose;
  height: 4.5rem;
  border: none;
  border-radius: 0.3rem;
  font-weight: 400;
}

.header-button:hover {
  background-color: $rose;
  color: $gray;
}
