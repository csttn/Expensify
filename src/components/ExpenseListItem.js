import React from "react";
import { Link } from "react-router-dom";
import moment from "moment";
import numeral from "./numeral";

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
  <Link className="list-item" to={`/edit/${id}`}>
    <div>
      <h2 className="list-item__title">{description}</h2>
      <span className="list-item__sub-title">
        {moment(createdAt)
          .locale("pt-br")
          .format("Do MMMM, YYYY")}
      </span>
    </div>
    <h3 className="list-item__data">
      {numeral(amount / 100).format("$  0,0.00")}
    </h3>
  </Link>
);

export default ExpenseListItem;
