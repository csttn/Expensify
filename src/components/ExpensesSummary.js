import React from "react";
import { connect } from "react-redux";
import numeral from "./numeral";
import selectExpenses from "../selectors/expenses";
import selectExpensesTotal from "../selectors/expenses-total";
import { Link } from "react-router-dom";

export const ExpensesSummary = ({ expenseCount, expenseTotal }) => {
  const expenseWord = expenseCount === 1 ? "despesa" : "despesas";
  const formattedExpenseTotal = numeral(expenseTotal / 100).format("$ 0,0.00");

  return (
    <div className="page-header">
      <div className="content-container">
        <div className="page-container">
          <h1 className="page-header__title">
            {/* Visualizando <span>{expenseCount}</span> {expenseWord},  */}
            Total: <span>{formattedExpenseTotal} </span>
          </h1>
          <div className="page-header__actions">
            <Link className="page-header__button" to="/create">
              <button className="page-header__button">
                Adicione uma nova despesa
              </button>
            </Link>
          </div>
        </div>
      </div>
    </div>
  );
};

const mapStateToProps = state => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expenseCount: visibleExpenses.length,
    expenseTotal: selectExpensesTotal(visibleExpenses)
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
