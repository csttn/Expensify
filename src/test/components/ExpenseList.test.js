import React from "react";
import { shallow } from "enzyme";
import { ExpenseList } from "../../components/ExpenseList";
import expenses from "../fixtures/expenses";

test("Renderizando ExpenseList com expenses", () => {
  const wrapper = shallow(<ExpenseList expenses={expenses} />);
  expect(wrapper).toMatchSnapshot();
});

test("Renderizando ExpenseList sem expenses", () => {
  const wrapper = shallow(<ExpenseList expenses={[]} />);
  expect(wrapper).toMatchSnapshot();
});
