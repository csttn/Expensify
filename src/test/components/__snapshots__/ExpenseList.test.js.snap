// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderizando ExpenseList com expenses 1`] = `
<div>
  <ExpenseListItem
    amount={195}
    createdAt={345601000}
    description="Cleyton"
    id={0}
    key="0"
    note=""
  />
  <ExpenseListItem
    amount={1345}
    createdAt={"1970-01-01T00:00:00.000Z"}
    description="julia"
    id={1}
    key="1"
    note=""
  />
  <ExpenseListItem
    amount={5}
    createdAt={864000500}
    description="Antonia"
    id={2}
    key="2"
    note=""
  />
  <ExpenseListItem
    amount={1}
    createdAt={172800500}
    description="Rafa"
    id={3}
    key="3"
    note=""
  />
</div>
`;

exports[`Renderizando ExpenseList sem expenses 1`] = `
<div>
  <p>
    no expenses
  </p>
</div>
`;
